number_test.go

```diff
@@ -5,6 +5,35 @@ import (
 	"testing"
 )
 
+func TestNumber_subtypes(t *testing.T) {
+	type (
+		MyInt     int
+		MyInt8    int8
+		MyInt16   int16
+		MyInt32   int32
+		MyInt64   int64
+		MyUint    uint
+		MyUint8   uint8
+		MyUint16  uint16
+		MyUint32  uint32
+		MyUint64  uint64
+		MyFloat32 float32
+		MyFloat64 float64
+	)
+	Number(t, MyInt(42)).EqualTo(42)
```

Comment:
I'm assuming these will generally be read from existing variables and fields how do you see these being used in a larger example?

Reply:
Sorry nevermind I see your intent now.